port/esp32/adapter/src/ipadapter.c

```diff
@@ -1731,8 +1731,12 @@ oc_dns_lookup(const char *domain, oc_string_t *addr, transport_flags flags)
 #endif /* OC_DNS_CACHE */
       freeaddrinfo(result);
     } else {
+#ifdef OC_DEBUG
       OC_ERR("failed to resolve address(%s) with error(%d): %s", domain, ret,
              gai_strerror(ret));
```

Comment:
those standard strerror/gai_strerror are usually pretty OK, why don't we want it for a non-OC_DEBUG build?

Reply:
because the symbol exist only in debug build.

Reply:
indeed, seems like some sort ESP32 peculiarity